eradiate/util/attrs.py

```diff
@@ -460,3 +460,22 @@ def f(instance, attribute, value):
             validator(instance, attribute, value)
 
     return f
+
+def make_validator_nequal_other(other):
+    def validator_nequal_other(instance, attribute, value):
+        """Validates if the validated value is different from the specfied
+        'other' value of the same class instance. Raises an AttributeError
+        if not.
+        This validator should only be used for arguments that can be cast
+        to numpy arrays.
+
+        Parameter ``other`` (attribute of instance)
+            Other attribute, which must be different from 'value'
+
+        """
+        other_value = getattr(instance, other)
+        if np.all(value == other_value):
+            raise AttributeError(f"'{attribute}' must be different "
+                                 f"from '{other}'!")
```

Comment:
`AttributeError` is not appropriate in such a case IMO (see _e.g._ [Python docs](https://docs.python.org/3/library/exceptions.html#AttributeError)). A `ValueError` is more appropriate. See [attrs docs](https://www.attrs.org/en/stable/api.html#attr.validators.in_) for examples of exceptions raised by validators.